automation/routine_publish_signals.go

```diff
@@ -111,64 +49,70 @@ type PublishSignals struct {
 	Transforms []func(item *views.ItemSave)
 }
 
-// Do performs the automation against c with the passed in opts.
-func (p PublishSignals) Do(ctx context.Context, c *clarify.Client, opts PublishOptions) error {
-	var total int
+var _ Routine = PublishSignals{}
+
+func (p PublishSignals) Do(ctx context.Context, cfg *Config) error {
+	logger := cfg.Logger()
+	client := cfg.Client()
+	earlyOut := cfg.EarlyOut()
```

Comment:
maybe add `dryRun := cfg.DryRun()` as well for consistency? Or remove `earlyOut` and use function directly